source/GameplayChangersSubstate.hx

```diff
@@ -94,6 +98,66 @@ class GameplayChangersSubstate extends MusicBeatSubstate
 
 		var option:GameplayOption = new GameplayOption('Botplay', 'botplay', 'bool', false);
 		optionsArray.push(option);
+
+		#if LUA_ALLOWED
+		var customGameplayChangers:Map<String, Bool> = new Map<String, Bool>();
+		var directories:Array<String> = [];
+
+		#if MODS_ALLOWED
+		directories.push(Paths.mods('custom_gamechangers/'));
+		directories.push(Paths.mods(Paths.currentModDirectory + '/custom_gamechangers/'));
+		for(mod in Paths.getGlobalMods())
+			directories.push(Paths.mods(mod + '/custom_gamechangers/'));
+		#end
+
+		for (i in 0...directories.length) {
+			var directory:String = directories[i];
+			if(FileSystem.exists(directory)) {
+				for (file in FileSystem.readDirectory(directory)) {
+					var path = haxe.io.Path.join([directory, file]);
+					if (!FileSystem.isDirectory(path) && file != 'readme.txt' && file.endsWith('.txt')) {
+						var fileToCheck:String = file.substr(0, file.length - 4);
+						var contentArray:Array<String> = File.getContent(path).split(",");
+						for(i in 0...contentArray.length)
+							if(contentArray[i].isSpace(0))
+								contentArray[i] = contentArray[i].substr(1, contentArray[i].length); // Remove space
+
+						var type:String = contentArray[0];
+						var tag:String = contentArray[1];
+						
+						if(!customGameplayChangers.exists(tag)) {
+							customGameplayChangers.set(tag, false);
+							contentArray.remove(type);
+							contentArray.remove(tag);
+							switch(type)
+							{
+								case 'int' | 'float' | 'percent':
+									var option:GameplayOption = new GameplayOption(fileToCheck, tag, type, contentArray[0]);
+									option.scrollSpeed = Std.parseFloat(contentArray[1]);
+									option.minValue = contentArray[2];
+									option.maxValue = contentArray[3];
+									option.changeValue = contentArray[4];
```

Comment:
Just to be safe, I think it's better to wrap these lines with Std.parseFloat() like below, instead of passing the string directly
```
option.minValue = Std.parseFloat(contentArray[2]);
option.maxValue = Std.parseFloat(contentArray[3]);
option.changeValue = Std.parseFloat(contentArray[4]);
```
There was a bug that I was facing when using Int values in the options and this seemed to fix it

Reply:
thanks for your suggestions, i have incorporated all your advice into the code